Assets/Scripts/UI/Login/UILoginMenu.cs

```diff
@@ -319,29 +319,37 @@ private static void PreventDeselection(ref GameObject previouslySelectedGameObje
         /// <returns>True if input should be handled, false if not.</returns>
         private static bool ShouldInputBeHandled()
         {
-            // TODO: Clarify why this is only evaluated for PS4 and PS5
-#if UNITY_PS4 || UNITY_PS5
-            // TODO: Simplify this conditional - it's sort of confusing to read despite doing exactly what we want
-            if (null == EventSystem.current || EventSystem.current.sendNavigationEvents == !EOSManager.Instance.IsOverlayOpenWithExclusiveInput())
-            {
-                return true;
-            }
-
             bool shouldHandle = !EOSManager.Instance.IsOverlayOpenWithExclusiveInput();
-                
-            Debug.Log($"Input {(shouldHandle ? "enabled" : "disabled")} due to EOS Overlay.");
+
 #if ENABLE_INPUT_SYSTEM
             EventSystem.current.currentInputModule.enabled = shouldHandle;
+            EventSystem.current.sendNavigationEvents = shouldHandle;
 #else
+            // TODO: Clarify why this is only evaluated for PS4 and PS5
+#if UNITY_PS4 || UNITY_PS5
             EventSystem.current.sendNavigationEvents = shouldHandle;
 #endif
+#endif
 
-            return shouldHandle;
-#else
-            return true;
+#if ENABLE_DEBUG_INPUT
+            LogInputChanged(shouldHandle);
 #endif
+            return shouldHandle;
+
         }
 
+#if ENABLE_DEBUG_INPUT
+        static bool previousShouldHandle = false;
+        static void LogInputChanged(bool shouldHandle)
+        {
+            bool result = previousShouldHandle != shouldHandle ? true : false;
+            if (result)
```

Comment:
I think it would be clearer if you changed this to be

```c#
if (previousShouldHandle != shouldHandle)
```
(including suggested changes from earlier comment)

That way you can entirely remove line `345`.

Reply:
yep, there used to be a `return result;`, then the function was rewritten into a void, so it is not needed anymore, nice find.